opentelemetry/src/common.rs

```diff
@@ -402,82 +402,55 @@ pub trait ExportError: std::error::Error + Send + Sync + 'static {
 
 /// Information about a library or crate providing instrumentation.
 ///
-/// An instrumentation library should be named to follow any naming conventions
+/// An instrumentation scope should be named to follow any naming conventions
 /// of the instrumented library (e.g. 'middleware' for a web framework).
 ///
 /// See the [instrumentation libraries] spec for more information.
 ///
 /// [instrumentation libraries]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/overview.md#instrumentation-libraries
 #[derive(Debug, Default, Clone)]
 #[non_exhaustive]
-pub struct InstrumentationLibrary {
+pub struct InstrumentationScope {
```

Comment:
this should be a relative easy renaming for users, so I don't think its worth maintaining the alias for few releases. 
This is part of `opentelemetry` and we are aiming to declare it RC in a week and stable soon after that.

Reply:
> this should be a relative easy renaming for users, so I don't think its worth maintaining the alias for few releases.
This is part of opentelemetry and we are aiming to declare it RC in a week and stable soon after that.

Yes, but in current form, this will break their instrumentation build, which is not good. We can still remove the deprecated code after reaching RC.

Reply:
> We can still remove the deprecated code after reaching RC.

You are right! However, this PR will break users pretty bad as it is not just renaming `InstrumentationLibrary`. I don't think its feasible to make this in a non-breaking way, and I feel its best to do this and unblock RC release of all APIs, so users know there won't be any further breaking change for instrumentations in the future.

(It might be possible to find ways to smoothen breaking change, but that effort might delay RC again and may not yield much value ultimately)

Reply:
From SIG meeting discussion:
1. We acknowledge the amount of breaking changes
2. We are delaying the 0.27 release by a week, to ensure we can finally declare `opentelemetry` crates logs, metrics, common as RC, to assure end users that the breaking changes are ending finally!
3. It is still technically possible to make breaking changes after RC, but the bar is extremely high.
4. We also should be declaring the `opentelemetry` metrics,logs,common things stable in the next ~1 month or so.
5. Because of this, its best to accept the cost of this breaking change now.